resources/files/windows/PuppetBolt/PuppetBoltFunctions.ps1

```diff
@@ -0,0 +1,1639 @@
+        [string]  $Username,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $Password,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $SSHPrivateKey,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $RunAs,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $SudoPassword,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [int]     $Concurrency,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [int]     $CompileConcurrency,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $ModulePath,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $BoltDir,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $ConfigFile,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $InventoryFile,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [bool]    $SaveRerun=$false,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $Transport,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [int]     $ConnectTimeout,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [bool]    $TTY=$false,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $Tmpdir,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [bool]    $Color=$true,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [BoltOutputLevel]  $OutputLevel
+    )
+    begin {
+        $_flags = Convert-BoltParams @PSBoundParameters
+    }
+    process {
+        return Invoke-BoltInternal -CommandString "script run $ScriptLocation" @PSBoundParameters
+    }
+}
+
+function Read-BoltPlans {
+    <#
+    .SYNOPSIS
+    Read all bolt plans available to Puppet Bolt
+    .DESCRIPTION
+    This function will execute 'bolt plan show <plan>' to list available plans or details
+    on a specific plan. Puppet Bolt (https://puppet.com/products/bolt) is an agentless automation
+    solution for running ad-hoc tasks and operations on remote targets
+    .PARAMETER PlanName
+    Name of the bolt plan to show details on
+    .PARAMETER ModulePath
+    List of directories containing modules, separated by ';'
+    .PARAMETER BoltDir
+    Specify what Boltdir to load config from (default: autodiscovered from current working dir)
+    .PARAMETER ConfigFile
+    Specify where to load config from (default: ~/.puppetlabs/bolt/bolt.yaml)
+    .PARAMETER Color
```

Comment:
Note as this is outputting structured data, the `Color` parameter is not necessary and will cause confusion.